source/ada/lsp-ada_handlers.adb

```diff
@@ -87,6 +96,27 @@ package body LSP.Ada_Handlers is
       return Location;
    end Get_Node_Location;
 
+   ---------------------------------
+   -- Send_Imprecise_Xref_Message --
+   ---------------------------------
+
+   procedure Send_Imprecise_Xref_Message
+     (Self     : access Message_Handler;
+      URI      : LSP.Messages.DocumentUri;
+      Position : LSP.Messages.Position;
+      Msg_Type : LSP.Messages.MessageType) is
+   begin
+      Self.Server.Show_Message
+        ((Msg_Type,
+         "Imprecise fallback used to compute cross-references on entity at:"
+         & To_LSP_String
+           (ASCII.LF & "   "  & To_UTF_8_String (URI))
```

Comment:
(minor) how  about we just send the base name of the file here,  to make the message less verbose?

Reply:
True, let's do that

Reply:
Actually I don't know how to convert the URI to a base name in a proper way... Do you know how to do it by chance?

Reply:
I would use URIs.File_Conversions to convert to a VFS.Virtual_File first, then use Display_Base_Name